### PR TITLE
Fix upload unknown STIX indicator

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17290

## Description
Fixes a bug where a legit STIX file added with unknown indicator would throw an error and fail. Now it'll ignore unknown indicators.
added unitest to test the bug.

## Does it break backward compatibility?
   - No

## Must have
- [X] Tests
- [ ] Code Review# Additional changes

### CODE REVIEW

**Comment by idobloch** [comment]
The stix2.json looks good

File: Scripts/StixParser/StixParser.py [hunk 1]
```diff
@@ -184,6 +184,21 @@ def extract_indicators(data):
...
+        Args:
+            indicator (dict): STIX-formatted entry
+        """
+        pattern = indicator.get("pattern")
+        if pattern:
+            groups = regex.findall(pattern)
+            if groups:
+                for key, value in patterns_dict.items():
+                    for term in groups:
+                        if len(term) == 2 and key in term[0]:
```
Comment (Itay4) [inline Scripts/StixParser/StixParser.py]: add comment to explain the logic

File: Scripts/StixParser/StixParser.py [hunk 2]
```diff
@@ -184,6 +184,21 @@ def extract_indicators(data):
         dict: containing all indicators data
 
     """
+    def get_indicators(indicator):
```
Comment (Itay4) [inline Scripts/StixParser/StixParser.py]: take it out and add test for it

**Comment by jochman** [comment]
after testing with @idobloch we found a bug that created duplicates indicator.
not fixed.

File: Scripts/StixParser/StixParser.py [hunk 3]
```diff
@@ -128,61 +128,44 @@ def dscore(score):
     return 0
 
 
-def build_entry(stx_obj, indicators):
+def build_entry(indicators_dict, stix_indicators_dict, pkg_id):
```
Comment (Itay4) [inline Scripts/StixParser/StixParser.py]: add return to docstring

File: Scripts/StixParser/StixParser.py [hunk 4]
```diff
@@ -128,61 +128,44 @@ def dscore(score):
...
+                )
+
+                result = create_indicator_entry(
+                    indicator_type=key,
+                    value=indicator,
+                    pkg_id=pkg_id,
+                    ind_id=ind_id,
+                    timestamp=timestamp,
+                    source=source,
+                    score=score,
```
Comment (Itay4) [inline Scripts/StixParser/StixParser.py]: remove trailing comma

File: Scripts/StixParser/StixParser.py (hunk 3 above)
Comment (jochman) [reply Scripts/StixParser/StixParser.py]: outdated (wrong line marked)

File: Scripts/StixParser/StixParser.py (hunk 2 above)
Comment (jochman) [reply Scripts/StixParser/StixParser.py]: problem with taking it out. anyway, it'll be used only in this function. you can use this function with sending input to the parent function (which only checks if we got dict or list

**Pull request merged**